client/state/login/reducer.js

```diff
@@ -38,10 +42,11 @@ export const requestSuccess = createReducer( null, {
 
 export const twoFactorAuth = createReducer( null, {
 	[ LOGIN_REQUEST ]: () => null,
-	[ LOGIN_REQUEST_SUCCESS ]: ( state, { data } ) => data || null,
+	[ LOGIN_REQUEST_SUCCESS ]: ( state, { data, rememberMe } ) => data ? { ...data, remember_me: rememberMe } : null,
 	[ TWO_FACTOR_AUTHENTICATION_LOGIN_REQUEST_FAILURE ]: ( state, { twoStepNonce } ) => Object.assign( {}, state, {
 		two_step_nonce: twoStepNonce
 	} ),
+	[ LOGIN_TWOFACTOR_UPDATE_NONCE ]: ( state, { twoStepNonce } ) => ( { ...state, two_step_nonce: twoStepNonce } ),
```

Comment:
Yep I don't think we have much choice but to keep those 2 actions. You could refactor this bit in it's own named reducer to reuse some code like @gziolo suggested.